Core/src/org/sleuthkit/autopsy/discovery/ui/MiniTimelineWorker.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Autopsy
+ *
+ * Copyright 2020 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.discovery.ui;
+
+import org.sleuthkit.autopsy.discovery.search.MiniTimelineResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import javax.swing.SwingWorker;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.discovery.search.DiscoveryEventUtils;
+import org.sleuthkit.autopsy.discovery.search.DomainSearch;
+
+/**
+ * SwingWorker to retrieve a list of artifacts for a specified type and domain.
+ */
+class MiniTimelineWorker extends SwingWorker<List<MiniTimelineResult>, Void> {
+
+    private final static Logger logger = Logger.getLogger(MiniTimelineWorker.class.getName());
+    private final String domain;
+
+    /**
+     * Construct a new ArtifactsWorker.
+     *
+     * @param artifactType The type of artifact being retrieved.
+     * @param domain       The domain the artifacts should have as an attribute.
+     */
+    MiniTimelineWorker(String domain) {
+        this.domain = domain;
+    }
+
+    @Override
+    protected List<MiniTimelineResult> doInBackground() throws Exception {
+        if (!StringUtils.isBlank(domain)) {
+            DomainSearch domainSearch = new DomainSearch();
+            return domainSearch.getAllArtifactsForDomain(Case.getCurrentCase().getSleuthkitCase(), domain);
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    protected void done() {
+        List<MiniTimelineResult> results = null;
+        if (!isCancelled()) {
+            try {
+                results = get();
+            } catch (InterruptedException | ExecutionException ex) {
+                logger.log(Level.SEVERE, "Exception while trying to get list of artifacts for Domain details for mini timeline view for domain: " + domain, ex);
```

Comment:
I don't think it should at this point, since we only want to entirely ignore things if the swing worker is cancelled in which case there will be a cancellation exception instead of an interruptedexception.

Reply:
I think the CancellationException is thrown if the get method is called after the swing worker has been cancelled and the InterruptedException is thrown if a waiting thread receives an interrupt during cancellation based on https://docs.oracle.com/javase/tutorial/uiswing/concurrency/cancel.html and https://docs.oracle.com/javase/7/docs/api/javax/swing/SwingWorker.html#get().  I'm not seeing anything in your get method that would directly throw an InterruptedException so it is probably okay, but if that get method was changed to use, say the cache or a sql query directly where an InterruptedException could potentially be thrown, it might be good to treat as a cancellation.